web/src/screens/private/Schedules/ViewScheduleDay.js

```diff
@@ -7,32 +7,218 @@ import moment from 'moment';
 
 import { withStyles } from 'material-ui/styles';
 import { CircularProgress } from 'material-ui/Progress';
-import Typography from 'material-ui/Typography';
 import Paper from 'material-ui/Paper';
 
+import styles from '../../../styles/AppStyle';
+
+import DayWeek from './components/DayWeek';
+import ScheduleHeader from './components/ScheduleHeader';
+import TableNextPrevious from '../../../components/Tables/TableNextPrevious';
+import SpreadPanel from '../../../components/Panels/SpreadPanel';
+import ScheduleDay from './components/ScheduleDay';
+import TimeRangeModal from '../../../components/Modals/TimeRangeModal';
+
 import SCHEDULE_QUERY from '../../../graphql/schedule.query';
-import styles from './ViewScheduleDay.styles';
-
-const ViewScheduleDay = ({ classes, loading, schedule, match }) =>
-  (loading ? (
-    <CircularProgress className={classes.progress} size={50} />
-  ) : (
-    <div className={classes.root}>
-      <Paper className={classes.rootPaper}>
-        <Typography variant="title" className={classes.paperTitle} gutterBottom>
-          {schedule.name} - ({moment.unix(match.params.time).format('LL')})
-        </Typography>
-      </Paper>
-    </div>
-  ));
+import {
+  CREATE_TIME_SEGMENT_MUTATION,
+  REMOVE_TIME_SEGMENT_MUTATION,
+  UPDATE_TIME_SEGMENT_MUTATION,
+} from '../../../graphql/time-segment.mutation';
+
+import { STATUS_AVAILABLE } from '../../../config';
+import { DEFAULT_START_TIME, DEFAULT_END_TIME } from '../../../constants';
+
+class ViewScheduleDay extends React.Component {
+  state = {
+    modalOpen: false,
+    modalTimeSegmentId: 0,
+    modalTitle: '',
+    modalUser: undefined,
+    modalStatus: STATUS_AVAILABLE,
+    modalStartTime: DEFAULT_START_TIME,
+    modalEndTime: DEFAULT_END_TIME,
+  };
+
+  onNextDateRange = () => {
+    const currentDay = moment.unix(parseInt(this.props.match.params.time, 10));
+    currentDay.add(1, 'days');
+    const { history } = this.props;
+    history.push(`/schedules/${this.props.match.params.id}/${currentDay.unix()}`);
+  };
+
+  onPreviousDateRange = () => {
+    const currentDay = moment.unix(parseInt(this.props.match.params.time, 10));
+    currentDay.add(-1, 'days');
```

Comment:
isnt that called moment().subtract ?

Reply:
Same thing - it's how it's done in .NET using add I think (old habit)